src/action/index.ts

```diff
@@ -64,6 +64,17 @@ async function main(): Promise<void> {
   }
 }
 
+/**
+ * Set output with logging to stdout for test support
+ */
+function setOutput(...args: Parameters<typeof setOutputActionsCore>) {
+  if (process.env.NODE_ENV === "test") {
+    console.log(`TEST::set-output name=${args[0]}::${args[1]}`);
+    return;
+  }
```

Comment:
I think this is acceptable for now, but moving forward I would like to restructure this to using a mocking library like [testdouble](https://github.com/testdouble/testdouble.js) rather than adding test-specific behavior to the actual source code

Reply:
Yes, I agree but was just being piratical about getting things working again with the existing patterns.